misk/src/main/kotlin/misk/client/ClientMetricsInterceptor.kt

```diff
@@ -110,8 +142,26 @@ class ClientMetricsInterceptor private constructor(
       labelNames = listOf("action", "code")
     )
 
-    fun create(clientName: String) = ClientMetricsInterceptor(clientName, requestDuration, requestDurationHistogram)
+    fun create(clientName: String) = ClientMetricsInterceptor(
+      clientName,
+      requestDuration,
+      requestDurationHistogram
+    )
   }
+
+  private class TrailerAwareResponseBody(
+    private val responseBody: ResponseBody,
+    private val func: () -> Unit
+  ) : ResponseBody() {
+    override fun contentLength() = responseBody.contentLength()
+    override fun contentType() = responseBody.contentType()
+    override fun source() = responseBody.source()
```

Comment:
I've never used it myself, but I think you should be able to use the kotlin delegation pattern (https://kotlinlang.org/docs/delegation.html) here. Then you wouldn't need these overrides that just delegate to `responseBody` as the compiler would generate them for you

Reply:
That's neat! But looks like it doesn't work with abstract classes, only for interfaces. I still renamed the property to convey the idea.